packages/cli/src/environments.ee/source-control/source-control-export.service.ee.ts

```diff
@@ -241,19 +247,38 @@ export class SourceControlExportService {
 				};
 			}
 
+			const allowedProjects =
+				await this.sourceControlScopedService.getAdminProjectsFromContext(context);
+
 			const fileName = getFoldersPath(this.gitFolder);
+
+			const existingFolders = await readFoldersFromSourceControlFile(fileName);
+
+			// keep all folders that are not accessible by the current user
+			// if allowedProjects is undefined, all folders are accessible by the current user
+			const newFolders =
```

Comment:
I think this is confusing: variable is called newFolders but contains a subset of existing folders. 
How about renaming it something like `foldersToKeepUnchanged` (or a shorter name if you have an idea). 
Then we could say
const newFolders = foldersToKeepUnchanged.concat(folders).map(...)